nomos-services/sdp/src/lib.rs

```diff
@@ -21,57 +22,77 @@ const BROADCAST_CHANNEL_SIZE: usize = 128;
 pub type FinalizedBlockUpdateStream =
     Pin<Box<dyn Stream<Item = FinalizedBlockEvent> + Send + Sync + Unpin>>;
 
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct SdpSettings {
+    /// Declaration ID for this node (set after posting declaration and
+    /// restarting)
+    pub declaration_id: Option<DeclarationId>,
+}
+
 pub enum SdpMessage {
     ProcessNewBlock,
     ProcessLibBlock,
     Subscribe {
         result_sender: oneshot::Sender<FinalizedBlockUpdateStream>,
     },
+
+    PostDeclaration {
+        service_type: ServiceType,
+        locators: Vec<Locator>,
+        reply_channel: oneshot::Sender<Result<DeclarationId, DynError>>,
+    },
+    PostActivity {
+        metadata: Vec<u8>, // DA/Blend specific metadata
+    },
+    PostWithdrawal {
+        declaration_id: DeclarationId,
+    },
 }
 
-pub struct SdpService<Backend: SdpBackend + Send + Sync + 'static, Metadata, RuntimeServiceId>
-where
-    Metadata: Send + Sync + 'static,
-{
+pub struct SdpService<Backend: SdpBackend + Send + Sync + 'static, RuntimeServiceId> {
     backend: PhantomData<Backend>,
     service_resources_handle: OpaqueServiceResourcesHandle<Self, RuntimeServiceId>,
     finalized_update_tx: broadcast::Sender<FinalizedBlockEvent>,
+    _current_declaration_id: Option<DeclarationId>,
 }
 
-impl<Backend, Metadata, RuntimeServiceId> ServiceData
-    for SdpService<Backend, Metadata, RuntimeServiceId>
+impl<Backend, RuntimeServiceId> ServiceData for SdpService<Backend, RuntimeServiceId>
 where
```

Comment:
I think this bound is not necessary at all, and we could then also remove it from hte `SdpService` struct declaration.